evals/elsuite/audio/eval.py

```diff
@@ -107,49 +107,54 @@ def do_completion(self, prompt, **kwargs):
 
 
 class MatchAudioTask(AudioTask):
-    def get_match_events(self):
+    def _get_match_events(self):
         return self.recorder.get_events("match")
 
-    def get_expected_values(self):
-        return [e.data["expected"] for e in self.get_match_events()]
+    def _get_expected_values(self):
+        return [e.data["expected"] for e in self._get_match_events()]
 
-    def get_sampled_values(self):
-        return [e.data["sampled"] for e in self.get_match_events()]
+    def _get_sampled_values(self):
+        return [e.data["sampled"] for e in self._get_match_events()]
 
-    def compute_corpus_metrics(self):
-        return {"accuracy": evals.metrics.get_accuracy(self.get_match_events())}
+    def _compute_corpus_metrics(self):
+        return {"accuracy": evals.metrics.get_accuracy(self._get_match_events())}
 
 
 class ModelGradedAudioTask(AudioTask):
     def __init__(
         self,
         completion_fns: list[CompletionFn],
         dataset: str,
+        eval_type: str,
         eval_completion_fn: CompletionFn,
         modelgraded_spec: str,
         modelgraded_spec_args: Optional[dict[str, dict[str, str]]] = None,
         *args,
         **kwargs,
     ):
         super().__init__(completion_fns, dataset, *args, **kwargs)
+        self.eval_type = eval_type
         self.eval_completion_fn = evals.registry.Registry().make_completion_fn(eval_completion_fn)
         self.eval_completion_kwargs = {"max_tokens": 1024}
         self.mg = self.registry.get_modelgraded_spec(modelgraded_spec)
         self.modelgraded_spec_args = modelgraded_spec_args or {}
 
-    def compute_metrics(self, sample: Sample, sampled: str):
+    def _compute_metrics(self, sample: Sample, sampled: str):
         completions = {"completion": sampled}
-        test_sample = {"input": self.build_prompt(sample, text_only=True)}
-        # TODO(juberti): this needs to call get_sample to get the GT answer...
+        test_sample = {
+            "input": self._build_prompt(sample, text_only=True),
+            "ideal": self._get_expected(sample),
+        }
         choice, info = classify(
             mg=self.mg,
+            eval_type=self.eval_type,
             completion_fn=self.eval_completion_fn,
             completion_kwargs=self.eval_completion_kwargs,
             format_kwargs={**completions, **test_sample, **self.modelgraded_spec_args},
         )
-        evals.record.record_metrics(choice=choice, score=info["score"])
+        evals.record.record_metrics(choice=choice, score=1.0 if choice in "BCE" else 0.0)
```

Comment:
What's BCE?

Reply:
Will add comment to this effect:

B = Answer is a superset of ground truth answer, C = answer matches GT answer, E = answer deviates from GT in unimportant ways.

A = Answer is subset, D = answer does not match GT.

Reply:
Oh I didn't know. It might be worth commenting this here as well, but you know best.